flink-doris-connector/src/main/java/org/apache/doris/flink/tools/cdc/DatabaseSync.java

```diff
@@ -250,8 +288,42 @@ public static class TableNameConverter implements Serializable {
             this.suffix = suffix == null ? "" : suffix;
         }
 
+        TableNameConverter(String prefix, String suffix,Pattern multiToOnePattern,Map<Pattern, String> multiToOneRulesPattern) {
+            this.prefix = prefix == null ? "" : prefix;
+            this.suffix = suffix == null ? "" : suffix;
+            this.multiToOnePattern=multiToOnePattern;
+            this.multiToOneRulesPattern = multiToOneRulesPattern;
+        }
+
         public String convert(String tableName) {
+            if(isMultiToOneTable(tableName)){
+                return getMultiToOneTargetByTableName(tableName);
+            }
```

Comment:
Is it possible to directly determine whether multiToOneRulesPattern is empty? If not, execute getMultiToOneTargetByTableName.